pkg/tools/etcd_helper.go

```diff
@@ -484,24 +484,14 @@ func (h *EtcdHelper) SetObj(key string, obj, out runtime.Object, ttl uint64) err
 	return err
 }
 
-// ResponseMeta contains information about the etcd metadata that is associated with
-// an object. It abstracts the actual underlying objects to prevent coupling with etcd
-// and to improve testability.
-type ResponseMeta struct {
-	// TTL is the time to live of the node that contained the returned object. It may be
-	// zero or negative in some cases (objects may be expired after the requested
-	// expiration time due to server lag).
-	TTL int64
-}
-
 // Pass an EtcdUpdateFunc to EtcdHelper.GuaranteedUpdate to make an etcd update that is guaranteed to succeed.
 // See the comment for GuaranteedUpdate for more detail.
-type EtcdUpdateFunc func(input runtime.Object, res ResponseMeta) (output runtime.Object, ttl *uint64, err error)
+type EtcdUpdateFunc func(input runtime.Object, node *etcd.Node) (output runtime.Object, ttl *uint64, err error)
```

Comment:
Conversation lost to the mists of github history, but basically: introduces coupling and encourages bad patterns - generic etcd should be generally isolated from exact etcd objects for test ability, and also callers should not be encouraged to accidentally (or intentionally) start using the node value directly